crates/bevy_ecs/macros/src/component.rs

```diff
@@ -595,40 +596,32 @@ impl Parse for Require {
 
         let func = if input.peek(Token![=]) {
             // If there is an '=', then this is a "function style" require
-            let _t: syn::Token![=] = input.parse()?;
+            input.parse::<Token![=]>()?;
             let expr: Expr = input.parse()?;
-            let tokens: TokenStream = quote::quote! (|| #expr).into();
-            Some(TokenStream2::from(tokens))
+            Some(quote!(|| #expr ))
         } else if input.peek(Brace) {
             // This is a "value style" named-struct-like require
             let content;
             braced!(content in input);
             let content = content.parse::<TokenStream2>()?;
-            let tokens: TokenStream = quote::quote! (|| #path { #content }).into();
-            Some(TokenStream2::from(tokens))
+            Some(quote!(|| #path { #content }))
         } else if input.peek(Paren) {
             // This is a "value style" tuple-struct-like require
             let content;
             parenthesized!(content in input);
-            let content = content.parse::<TokenStream2>()?;
+            let content = content.parse_terminated(Expr::parse, Token![,])?;
```

Comment:
I'd prefer to use the "passthrough" approach here too for consistency and performance.